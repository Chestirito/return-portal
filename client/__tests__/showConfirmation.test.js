import React from "react";
import ReactDOM from 'react-dom';
import ConfirmationPage from '../pages/Confirmation/showConfirmation';
import "@testing-library/jest-dom/extend-expect";
import {render, fireEvent} from '@testing-library/react';
import 'core-js';
import renderer from 'react-test-renderer';
require("regenerator-runtime/runtime");
global.fetch = require('jest-fetch-mock');

let props = {
    serveoname:"localhost:9000",
    code: "kdfbgk2985",
    email: "randomReturn@email.com"
}

function renderComp(){
    const utils = render(<ConfirmationPage
                            {...props}
                        />);
    return utils;
}

beforeEach(()=>{
    window.alert = jest.fn();
    
    // fetch.mockResponse(JSON.stringify({
    //     product:{
    //         image:{
    //             src: "imageSourceURL"
    //         }
    //     }
    // }))
})

it("renders without issue",()=>{
    const tree = renderer
                    .create(<ConfirmationPage
                                {...props}
                            />)
                    .toJSON();
    expect(tree).toMatchSnapshot();
})

it("resend email", ()=>{
    const spy = jest.spyOn(ConfirmationPage.prototype, "sendEmail");
    const {getByText} = renderComp();
    fireEvent.click(getByText(/resend email/i));
    expect(spy).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledTimes(1);
})

it("do not resend email if succesive click", ()=>{
    const spyAlert = jest.spyOn(window, "alert");
    const {getByText} = renderComp();
    fireEvent.click(getByText(/resend email/i));
    fireEvent.click(getByText(/resend email/i));
    expect(spyAlert).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledTimes(1);
})

