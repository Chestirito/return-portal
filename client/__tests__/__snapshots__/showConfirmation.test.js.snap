// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without issue 1`] = `
<div>
  <div
    className="centre"
  >
    <h2
      className="confirmCode"
    >
      Your confirmation code is: 
      <strong>
        kdfbgk2985
      </strong>
    </h2>
    <h3
      className="confirmTitle"
    >
      Thank you for your return request.
    </h3>
    <p
      className="explanation"
    >
      A confirmation email has been sent to 
      randomReturn@email.com
      . To make sure that we can process your return as quickly as possible, please follow these steps when returning your package.
    </p>
    <button
      className="Submit2"
      onClick={[Function]}
    >
      Resend email
    </button>
  </div>
  <br />
  <div
    className="confirmation1"
  >
    <h2
      className="confirmation"
    >
      Item Drop-off Instructions
    </h2>
    <div
      className="instructions"
    >
      <br />
      <p>
        Request return on Flindel and receive confirmation code (e.g. A1B2C3)
      </p>
      <p>
        Present your code and give your return item(s) to one of the staff at one of the locations on the map. Look for someone wearing a red shirt carrying a big bag!
      </p>
      <p>
        Wait for your refund!
      </p>
    </div>
  </div>
  <div
    className="confirmation2"
  >
    <h2
      className="confirmation"
    >
      Drop-Off Locations
    </h2>
    <div
      className="dropLocations"
    >
      <div
        className="mapContainer"
      />
      <div
        className="dropAdd"
      >
        <p
          className="dropAddress"
        >
          1. 123 Dundas Street (Starbucks)
        </p>
        <p
          className="dropAddress"
        >
          2. 987 Bathurst Street (Tim Hortons)
        </p>
        <p
          className="dropAddress"
        >
          3. 456 Front Street (Library) 
        </p>
        <p
          className="dropAddress"
        >
          4. 159 Wilson Street (Starbucks)
        </p>
        <p
          className="dropAddress"
        >
          5. 354 Bay Street (Store)
        </p>
      </div>
    </div>
  </div>
  <div
    className="customerService"
  >
    <p>
      If you have any questions, 
      <br />
      please contact us at
       
      <a
        href="mailto:customerservice@flindel.com"
      >
        customerservice@flindel.com
      </a>
    </p>
  </div>
</div>
`;
